src/transformers/models/pop2piano/tokenization_pop2piano.py

```diff
@@ -0,0 +1,508 @@
+                notes.append([note_onset, offset_idx, pitch, self.encoder["DEFAULT_VELOCITY"]])
+
+        if len(notes) == 0:
+            return []
+        else:
+            notes = np.array(notes)
+            note_order = notes[:, 0] * 128 + notes[:, 1]
+            notes = notes[note_order.argsort()]
+            return notes
+
+    def notes_to_midi(self, notes: np.ndarray, beatstep: np.ndarray, offset_sec: int = 0.0):
+        """
+        Converts notes to Midi.
+
+        Args:
+            notes (`numpy.ndarray`):
+                This is used to create Pretty Midi objects.
+            beatstep (`numpy.ndarray`):
+                This is the extrapolated beatstep that we get from feature extractor.
+            offset_sec (`int`, *optional*, defaults to 0.0):
+                This represents the offset seconds which is used while creating each Pretty Midi Note.
+        """
+
+        requires_backends(self, ["pretty_midi"])
+
+        new_pm = pretty_midi.PrettyMIDI(resolution=384, initial_tempo=120.0)
+        new_inst = pretty_midi.Instrument(program=0)
+        new_notes = []
+
+        for onset_idx, offset_idx, pitch, velocity in notes:
+            new_note = pretty_midi.Note(
+                velocity=velocity,
+                pitch=pitch,
+                start=beatstep[onset_idx] - offset_sec,
+                end=beatstep[offset_idx] - offset_sec,
+            )
+            new_notes.append(new_note)
+        new_inst.notes = new_notes
+        new_pm.instruments.append(new_inst)
+        new_pm.remove_invalid_notes()
+        return new_pm
+
+    def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] = None) -> Tuple[str]:
+        """
+        Saves the tokenizer's vocabulary dictionary to the provided save_directory.
+
+        Args:
+            save_directory (`str`):
+                A path to the directory where to saved. It will be created if it doesn't exist.
+            filename_prefix (`Optional[str]`, *optional*):
+                A prefix to add to the names of the files saved by the tokenizer.
+        """
+        if not os.path.isdir(save_directory):
+            logger.error(f"Vocabulary path ({save_directory}) should be a directory")
+            return
+
+        vocab_file = os.path.join(
+            save_directory, (filename_prefix + "-" if filename_prefix else "") + VOCAB_FILES_NAMES["vocab_file"]
+        )
+        with open(vocab_file, "w", encoding="utf-8") as f:
+            f.write(json.dumps(self.encoder))
+
+        return (vocab_file,)
+
+    def __call__(
+        self,
+        token_ids: Union[List, TensorType],
+        feature_extractor_output: BatchFeature,
+        return_midi: bool = True,
+    ):
+        r"""
+        This is the `__call__` method for `Pop2PianoTokenizer`. It converts the tokens generated by the transformer to
+        midi_tokens and returns them.
+
+        Args:
+            token_ids (`torch.LongTensor`):
```

Comment:
Actually it is supposed to take `np` or `tf` or `pt` or `jax`, since I dont want to invoke `pt` or `tf` here which will increase the Tokenizer's dependency, lets leave the type hint for `token_ids` blank, instead I am adding `Union[np.ndarray, torch.Tensor, tf.Tensor]` to the docstring just like [whisper tokenizers decode](https://github.com/huggingface/transformers/blob/9859806608c0609ff04cc7999cd04c5252fbc1fb/src/transformers/models/whisper/tokenization_whisper.py#L558) method.